sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -255,17 +255,21 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
             }
           }
 
-        case _ => throw new AnalysisException(s"$cls does not support data writing.")
+        case _ => saveToV1Source()
```

Comment:
This section got more complicated with this. Add more comments above this section on this fallback policy.